### PR TITLE
[SPARK-43576][CORE] Remove unused declarations from Core module

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to remove unused declarations from `Core` module

### Why are the changes needed?
Make code clean.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 1]
```diff
@@ -128,8 +128,6 @@ class HadoopRDD[K, V](
 
   protected val jobConfCacheKey: String = "rdd_%d_job_conf".format(id)
 
-  protected val inputFormatCacheKey: String = "rdd_%d_input_format".format(id)
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala:131]: This is a part of developer API. I don't think we can just remove this out.

**Comment by HyukjinKwon** [comment]
@panbingkun how did you find them?

FWIW, I thought we added a compilation feature to disallow unused variables IIRC, @LuciferYang .

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala (hunk 1 above)
Comment (panbingkun) [reply core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala:131]: Ok, Let's reserve it.

**Comment by panbingkun** [comment]
> @panbingkun how did you find them?
> 
> FWIW, I thought we added a compilation feature to disallow unused variables IIRC, @LuciferYang .

@HyukjinKwon 

1.When I review the HadoopRDD code, I noticed some inspection prompts, eg:
image

2.So I wonder if there are other cases in the core module, then I export `unused declaration` rule to single file, name as `unused_declaration.xml`
image

3.execute command:
"/Users/panbingkun/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/231.9011.34/IntelliJ IDEA CE.app/Contents/bin/inspect.sh" /Users/panbingkun/Developer/spark/spark-community /Users/panbingkun/Developer/spark/spark-community/unused_declaration.xml /Users/panbingkun/Developer/spark/spark-community/unused_declaration_results -v2 -d /Users/panbingkun/Developer/spark/spark-community/core/src

4.The declared unused variable results in all `core` modules will be saved in the `unused_declaration_results` file.

5.Open it by IDE.
image

6.Some inspect results may be inaccurate and ultimately require manual reconfirmation.

**Comment by LuciferYang** [comment]
> IIRC

Currently, we have only added compilation checks on `-Ywarn-unused:imports` for Scala 2.12 (the behavior of Scala 2.13 is slightly different). 

https://github.com/apache/spark/blob/411bcd2f18b2275466edc752c8cdcdfcaab1cb9c/pom.xml#L2864

For the current version of Scala 2.12, we can further try to add more checks, such as `patvars`, `privates`, and `locals`. For codes that cannot be changed, we can use `@nowarn` to suppress them.

File: core/src/main/resources/org/apache/spark/ui/static/executorspage.js [hunk 2]
```diff
@@ -126,7 +126,6 @@ function totalDurationAlpha(totalGCTime, totalDuration) {
     (Math.min(totalGCTime / totalDuration + 0.5, 1)) : 1;
 }
 
-// When GCTimePercent is edited change ToolTips.TASK_TIME to match
```
Comment (MaxGekk) [inline core/src/main/resources/org/apache/spark/ui/static/executorspage.js:129]: hmm, why do you propose to remove this?

Comment (panbingkun) [reply core/src/main/resources/org/apache/spark/ui/static/executorspage.js:129]: ToolTips.TASK_TIME - Shaded red when garbage collection (GC) time is over 10% of task time

- Adding this variable for the first time
CommitID: c037d25482ea63430fb42bfd86124c268be5a4a4 
JIRA: https://issues.apache.org/jira/browse/SPARK-12149
[SPARK-12149][WEB UI] Executor UI improvement suggestions - Color UI
PR: https://github.com/apache/spark/pull/10154
First declaration：https://github.com/apache/spark/pull/10154/files#diff-91f8d288f7aa0726fc067ad9eb67aff41c474ada4d702e5b393bd81ae5aee7b5R91-R92
First use：https://github.com/apache/spark/pull/10154/files#diff-3bb7f9962a6343cbc09be474b85f2f9ed2e80047f4863d447faa407d925f425cR82


- The variable content is directly hard coding in `executorspage-template.html` file
CommitID: b9bab4dcf6cec5ec9e9860871f9dd1ec633a1d22
JIRA: https://issues.apache.org/jira/browse/SPARK-15951
[SPARK-15951] Change Executors Page to use datatables to support sorting columns and searching
PR: https://github.com/apache/spark/pull/13670
The variable content is directly hard coding: https://github.com/apache/spark/pull/13670/files#diff-cc6993f75c03b53bb41d8d7113ed3d5157580e583672da68b15cbe4d9f374e15R36
Delete: https://github.com/apache/spark/pull/13670/files#diff-3bb7f9962a6343cbc09be474b85f2f9ed2e80047f4863d447faa407d925f425cL94


**So I think it needs to be removed; Meanwhile, I did not find any usage in the `spark` code base using the following command: find . -name "*" -type file -print0 | xargs -0 grep "TASK_TIME"**

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun and @LuciferYang @HyukjinKwon for review.

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM.

**Pull request closed without merging**